ssts/segmentation/example.py

```diff
@@ -54,28 +54,35 @@ def main():
     elif example_number == "2":
 
         # Initialize GMM segmentation
-        pre_seg = seg_gmm.SegmenterGMM(n_components=2, embedding_dim=3)
+        num_pca_components = 3
+        pre_seg = seg_gmm.SegmenterGMM(n_components=2, embedding_dim=num_pca_components)
 
         # Run segmentation
         pre_labels = pre_seg.fit_transform(data)
 
         # Plot classification
-        #mm.show_classification(pre_labels, data)
+        mm.show_classification(pre_labels, data)
 
         # Plot classification distributions
-        #mm.show_classification_distributions(pre_labels, data)
+        mm.show_classification_distributions(pre_labels, data)
+
+        # Handling PCA components from data
```

Comment:
The PCA model is already trained if running the GMM segmentation class with a specified embedding_dim. So you can access its components in the following manner.